### ISSUE TITLE
Test start script does nor work properly

### ISSUE THREAD

**Issue by franzlst** [DLR-RM/RAFCON#147]
After running a state machine with subprocess, the test script often cannot find the file created by the state machine.
See for example this log output of the BuildBot:
http://rmc-chimaere:8010/builders/common%2Frafcon_sled11-x86_64-gcc4.x/builds/328/steps/test/logs/stdio

```
    def test_start_script_open():
        script = dirname(realpath(rafcon.__file__)) + "/statemachine/start.py"
        start_path = rafcon.__path__[0] + "/../test_scripts/unit_test_state_machines/start_script_test"
        cmd = sys.executable + " %s -o %s" % (script, start_path)
        subprocess.call(cmd, shell=True)
>       tmp_file = open("/tmp/rafcon_unit_tests/test_start_script.txt", "r")
E       IOError: [Errno 2] No such file or directory: '/tmp/rafcon_unit_tests/test_start_script.txt'
```

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-08-01 06:35:46+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I have seen this and talked to Franz about this failure. First, I estimated that the state-machine is not finished and the file not created and it is caused by the execution or asynchronous call. 

But if we read the full failure message, see link from above you will find it is a import failure of the profiler while the subprocess.call is down:

```
=================================== FAILURES ===================================
____________________________ test_start_script_open ____________________________

    def test_start_script_open():
        script = dirname(realpath(rafcon.__file__)) + "/statemachine/start.py"
        start_path = rafcon.__path__[0] + "/../test_scripts/unit_test_state_machines/start_script_test"
        cmd = sys.executable + " %s -o %s" % (script, start_path)
        subprocess.call(cmd, shell=True)
>       tmp_file = open("/tmp/rafcon_unit_tests/test_start_script.txt", "r")
E       IOError: [Errno 2] No such file or directory: '/tmp/rafcon_unit_tests/test_start_script.txt'

source/test/common/test_start_script.py:16: IOError
----------------------------- Captured stdout call -----------------------------
15:41:33:    DEBUG - rafcon.statemachine.config:  Configuration loaded from /home/f_buildb/.config/rafcon/config.yaml
15:41:33:     INFO - rafcon.statemachine.config:  Config use default-config-file parameter 'PROFILER_RUN': False.
15:41:33:     INFO - rafcon.statemachine.config:  Config use default-config-file parameter 'PROFILER_VIEWER': True.
15:41:33:     INFO - rafcon.statemachine.config:  Config use default-config-file parameter 'PROFILER_RESULT_PATH': /tmp/rafcon_profiler_result.prf.
15:41:33:    DEBUG - rafcon.statemachine.library_manager:  Initializing Storage object ...
15:41:33:    DEBUG - rafcon.statemachine.execution.state_machine_status:  State machine status is set to STATE_MACHINE_EXECUTION_STATUS.STOPPED
15:41:33:    DEBUG - rafcon.statemachine.execution.state_machine_execution_engine:  State machine execution engine initialized
----------------------------- Captured stderr call -----------------------------
Xlib:  extension "RANDR" missing on display ":10.0".
Traceback (most recent call last):
  File "/home_local/f_buildb/buildbot_slave/buildbot_config/slave/build_data/common_rafcon_sled11-x86_64-gcc4_x/build/src/rafcon/source/rafcon/statemachine/start.py", line 33, in <module>
    from rafcon.utils import profiler
ImportError: cannot import name profiler
!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!
================= 1 failed, 1 pytest-warnings in 1.83 seconds ==================
```

As long as the test even not checks for failure of the shell command call the file may is missing. So I add the calls (see 4032ca8) and we have to find out why the import of the profiler in this case of a shell command fails.


Originally created by @Rbelder  at 2016-08-01 09:57:15+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
@Benno: May you have seen it, I checked if the import failure occur without shell=True by using subprocess.Popen but it does not change it. We should do the debug on the build-bot pc and with the same setup. I was using commits to do the tests and that is bad style.

Sorry, all. :)


Originally created by @Rbelder  at 2016-08-01 10:50:01+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Fixed in 71f542d842d4c2379e7a57407311c53e5cdae602

A subproccess gains the same environment as the calling process. The calling process, generated by BuildBot, creates the environment using RMPM and thus using the paths from the latest RAFCON release.

Something in the calling process probably adds .`[...]/source` to `sys.path`, where Python finds the latest _commit_ of RAFCON. However, `sys.path` is not part of the environment and is thus not copied to the subprocess. Therefore, the subprocess only find the latest _release_ of RAFCON in its environment. In this release, the profiler has not been moved.

I solved the problem by manually adding the RAFCON path to the subprocess' PYTHONPATH. A better solution would be to let BuildBot do this. I will think about this some time later :smile: 


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-08-01 11:29:49+00:00 (moved from RMC internal repository)
